operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -0,0 +1,409 @@
+		if !isDeepCopy {
+			ceb.ceb.ObjectMeta = srcCeb.ObjectMeta
+		} else {
+			srcCeb.DeepCopyInto(ceb.ceb)
+			for _, cep := range ceb.ceb.Endpoints {
+				// Update the cacheCepMap, to reflect all CEPs are packed in a CEB
+				c.cacheCepMap.insert(getCepNameFromCCEP(&cep), srcCeb.GetName())
+			}
+		}
+	}
+}
+
+// If available, getCebFromCache returns CiliumEndpointBatch object.
+func (c *cebManagerFcfs) getCebFromCache(cebName string) (*cilium_v2.CiliumEndpointBatch, error) {
+	if ceb, ok := c.cacheCeb[cebName]; ok {
+		outCeb := new(cilium_v2.CiliumEndpointBatch)
+		ceb.backendMutex.Lock()
+		ceb.ceb.DeepCopyInto(outCeb)
+		ceb.backendMutex.Unlock()
+		return outCeb, nil
+
+	}
+	return nil, fmt.Errorf("Failed to get CEB from local cache for the cebName: %s", cebName)
+}
+
+// findCeb returns the available ceBatch object and is chosen based on FCFS.
+// if all CEB's reached max capacity or marked for delete. Allocate a new CEB.
+func (c *cebManagerFcfs) findCeb() (ceb *ceBatch) {
+
+	// get first available CEB
+	for _, ceb = range c.cacheCeb {
+		if len(ceb.ceb.Endpoints) >= c.maxCepsInCeb || len(ceb.ceb.Endpoints) == 0 {
+			continue
+		}
+		return
+	}
+
+	// allocate a new ceBatch and return
+	ceb = c.createCeb("")
+	return
+}
+
+// InsertCepInCache is used to Insert CEP in local cache, this may result in Creating a New
+// CEB object or Updating an existing CEB object.
+func (c *cebManagerFcfs) InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) (string, error) {
+
+	// check the given cep is already exists in any of the CEB.
+	// if yes, Update a ceb with the given cep object.
+	if cebName, exists := c.cacheCepMap.get(getCepNameFromCCEP(cep)); exists {
+		// add a cep into the ceb
+		addCEPtoCEB(cep, c.cacheCeb[cebName])
+		c.queue.Add(cebName)
+		return cebName, nil
+	}
+
+	// If given cep object isn't packed in any of the CEB. find a new ceb
+	// to pack this cep.
+	cb := c.findCeb()
+
+	// Cache CEP name with newly allocated CEB.
+	c.cacheCepMap.insert(getCepNameFromCCEP(cep), cb.ceb.GetName())
+
+	// Queue the CEP in CEB
+	addCEPtoCEB(cep, cb)
+	c.queue.Add(cb.ceb.GetName())
+	return cb.ceb.GetName(), nil
+}
+
+// RemoveCepFromCache is used to remove CEP from local cache, this may result in
+// Updating an existing CEB object.
+func (c *cebManagerFcfs) RemoveCepFromCache(cep *cilium_v2.CoreCiliumEndpoint) error {
+
+	cepName := getCepNameFromCCEP(cep)
+	log.WithFields(logrus.Fields{
+		"cep-name": cepName,
+	}).Debug("Remove CEP from local cache")
+
+	// Check in local cache, if a given cep is already batched in one of the ceb.
+	// and if exists, delete cep from ceb.
+	if cebName, exists := c.cacheCepMap.get(cepName); exists {
+		var ceb *ceBatch
+		if ceb, exists = c.cacheCeb[cebName]; !exists {
+			log.Info("Valid cep Mapping, but no matching CEP in CEB")
+			return nil
+		}
+		// Check for valid ceb object
+		if ceb == nil || ceb.ceb == nil {
+			log.Info("Valid cep Mapping, but not valid CEB")
```

Comment:
I had mentioned earlier also, during development phase,  we had operator crash because of invalid `ceb.ceb` or `ceb` pointer.  I don't know when and why his happened.

Reply:
Tried multiple times, Issue no longer reproducible.

Reply:
Then please remove this code unless you know why this would happen.

Reply:
Alright will do.